internal/command/prepare.go

```diff
@@ -27,19 +27,28 @@ func executePrepare(ctx context.Context, env gitlab.Environment) error {
 	log.SetOutput(os.Stderr)
 	log.Println("Running prepare stage")
 
-	if env.TemplateId == "" {
-		return fmt.Errorf("failed to get template id from environment: %w", gitlab.ErrMissingVar)
-	}
-
 	apiClientConfig := getAPIClientConfig(env)
 	apiClient, err := ankacloud.NewAPIClient(apiClientConfig)
 	if err != nil {
 		return fmt.Errorf("failed to initialize API client with config +%v: %w", apiClientConfig, err)
 	}
 	controller := ankacloud.NewController(apiClient)
 
+	templateId := env.TemplateId
+	if templateId == "" {
+		if env.TemplateName == "" {
+			return fmt.Errorf("no template ID or Name supplied: %w", gitlab.ErrMissingVar)
+		}
+		log.Println("Please consider using template ID instead of template name, since template names are not guaranteed to be unique")
+		templateId, err = controller.GetTemplateIdByName(ctx, env.TemplateName)
+		if err != nil {
+			return fmt.Errorf("failed to get template id of template named %q: %w", env.TemplateName, err)
+		}
+		log.Printf("template id for template name %q: %s\n", env.TemplateName, templateId)
```

Comment:
`template with ID %q and name %q will be used`

lets either make it capitalized as previous `log.Println` or both not capitalized